bots/pr/src/test/java/org/openjdk/skara/bots/pr/BackportTests.java

```diff
@@ -1347,4 +1347,115 @@ void badIssueInOriginal(TestInfo testInfo) throws IOException {
+            assertTrue(backportComment.contains("does not exist in project"));
+            assertFalse(pr.labelNames().contains("backport"));
+
+            // Use different kinds of good titles
+            // Use the full issue ID
+            pr.setTitle("Backport " + issue1.id());
+            TestBotRunner.runPeriodicItems(bot);
+            backportComment = pr.comments().get(2).body();
+            assertTrue(backportComment.contains("This backport pull request has now been updated with the original issue"));
+            assertEquals(issue1Number + ": An issue", pr.title());
+            assertTrue(pr.labelNames().contains("backport"));
+
+            // Set the title without project name
+            pr.setTitle("Backport " + issue1.id().split("-")[1]);
+            TestBotRunner.runPeriodicItems(bot);
+            backportComment = pr.comments().get(3).body();
+            assertTrue(backportComment.contains("This backport pull request has now been updated with the original issue"));
+            assertEquals(issue1Number + ": An issue", pr.title());
+            assertTrue(pr.labelNames().contains("backport"));
+
+            // Approve PR and re-run bot
+            var prAsReviewer = reviewer.pullRequest(pr.id());
+            prAsReviewer.addReview(Review.Verdict.APPROVED, "Looks good");
+            TestBotRunner.runPeriodicItems(bot);
+            assertLastCommentContains(pr, "This change now passes all *automated* pre-integration checks");
+
+            // Integrate
+            var prAsCommitter = author.pullRequest(pr.id());
+            prAsCommitter.addComment("/integrate");
+            TestBotRunner.runPeriodicItems(bot);
+
+            // Find the commit
+            assertLastCommentContains(pr, "Pushed as commit");
+
+            String hex = null;
+            var comment = pr.comments().get(pr.comments().size() - 1);
+            var lines = comment.body().split("\n");
+            var pattern = Pattern.compile(".* Pushed as commit ([0-9a-z]{40}).*");
+            for (var line : lines) {
+                var m = pattern.matcher(line);
+                if (m.matches()) {
+                    hex = m.group(1);
+                    break;
+                }
+            }
+            assertNotNull(hex);
+            assertEquals(40, hex.length());
+            localRepo.checkout(localRepo.defaultBranch());
+            localRepo.pull(author.url().toString(), "master", false);
+            var commit = localRepo.lookup(new Hash(hex)).orElseThrow();
+
+            var message = CommitMessageParsers.v1.parse(commit);
+            assertEquals(1, message.issues().size());
+            assertEquals("An issue", message.issues().get(0).description());
+            assertEquals(List.of("integrationreviewer3"), message.reviewers());
+            assertEquals(List.of(), message.contributors());
+            assertEquals(List.of(), message.summaries());
+            assertEquals(List.of(), message.additional());
```

Comment:
It might be good to also check that the commit message does not contain "Backport of", or is this covered by checking that `message.additional()` is empty?

Reply:
Good point. I just removed the assert for the existence of the original hash without thinking about inverting that assert.